### ISSUE TITLE
Line width of Diagram View

### ISSUE THREAD

**Issue by christiankral** [OpenModelica/OpenModelica#8217]
This is a feature request. 

In order to create high quality Modelica diagrams in different kinds of publications I am using the SVG export of the Diagram View in OpenModelica. The quality of the exported SVG file is very good. Yet, in many cases the publishers require a minimum line width of each drawing in order to overcome printing problems. 

However, the post processing of SVG files is still causing issues as described in [trac/#5645](https://trac.openmodelica.org/OpenModelica/ticket/5645), which may not be resolved easily.

An alternative approach were to implement a configuration for the line width of the Diagram view exports, similar to the line width of the plots lines: 

![image](https://user-images.githubusercontent.com/4184218/143244285-22f84ffa-d4f1-4415-8344-770326abf73e.png)

If the line width were also applied to the Diagram View on the screen, this could possibly also improve the quality of diagrams presented through a projector.

**Comment by casella** [comment]
Sounds good, let's see if we can do that in the next release

**Comment by adeas31** [comment]
The diagram uses a different set of settings `Tools-Options-Line Style->Thickness`. But these settings are only used when you are modeling your own models. If you are using models from some other libraries it depends on what thickness the library developer has used.

I believe what you want is to have a minimum line thickness when exporting diagrams (perhaps for all formats and not limited to SVG). If the line thickness is already greater than the minimum then use it otherwise enhance it to the minimum.

**Comment by christiankral** [comment]
Yes, this is exactly what I meant. It is about the minimum line thickness when exporting Diagram Views as SVG (and yes, also PNG) files.

**Comment by ceraolo** [comment]
> Yes, this is exactly what I meant. It is about the minimum line thickness when exporting Diagram Views as SVG (and yes, also PNG) files.

Omedit uses larger-width lines when connecting vector variables. 
Maybe what you are asking is better implemented as "use double width lines in diagrams?"
This would mean that vector connections will still be thicker than scalar, but all lines will be thicker than today, which improves the visual quality in some cases, e.g  on 4k screens

**Comment by casella** [comment]
@ceraolo you should also test it. Much improved rendering of diagrams now.

**Comment by ceraolo** [comment]
I checked my models.
Indeed I did not have any special issue on line witdth.
So my checks were mainly to see is any regression had occurred (because of my possible fancy usage of line widths).
Result: everything remained as it was, i.e. in good shape.

**Comment by christiankral** [comment]
This is a very good progress. Thank you @adeas31.

I tested the linewidth of the Diagram view with the actual nightly version. The exported SVN and PNG files behave the same which is very good.

I noticed two aspects that I want to mention:

1. When selecting a certain linewidth 

![image](https://github.com/OpenModelica/OpenModelica/assets/4184218/715bcb5d-b512-4b79-8fe6-62b415518e29)

the linewidths of newly openend Diagram views are displayed correctly. After picking a different linewidth, the Diagram view of a newly opened model ist shown with the different linewidth. The previously opened model, however, is still displayed with the previous linewidth setting. So apparently, the Diagram view is somehow cached and the displayed linewidth is not updated after changing it.

2. The linewidth is applied to connection lines only, but not the icons which are also rendered in the Diagram view:

![image](https://github.com/OpenModelica/OpenModelica/assets/4184218/72bf3901-3b5e-4edf-8a53-baaae52c0227)

Therefore, the diagram somehow looks inconsistently drawn. From my understanding, I would draw the icons of the Diagram view the selected linewidth, too.

**Comment by adeas31** [comment]
1. When you set line thickness in settings this means from that point on all the newly created shapes will take that line thickness. This will not change the line thickness of existing models nor does it change the line thickness of any model that you will load. It will only be applied to new classes that the user create.

2. If you create a model A and then draw an icon using line thickness of 1. And then you add this icon into another model B then the icon is scaled based on its coordinate system (default is 0.1 if nothing is specified). Draw connections from the icon and the connections will get whatever line width is in the settings lets say 1. The connections are not scaled so you see the width as it is but the icon is scaled so you don't see the width as 1.

**Comment by ceraolo** [comment]
> The diagram uses a different set of settings `Tools-Options-Line Style->Thickness`. But these settings are only used when you are modeling your own models. If you are using models from some other libraries it depends on what thickness the library developer has used.

I haven't not prepared a test yet, but it seems to me that thickness is not managed as expected by OM (and Dymola). 
The default thickness for lines is 0.25 that, if I unserstand well, means 0.25 mm. But on very high reslolution screens, i.e. 3k or 4k, connecting lines with default thickness are displayed bery, very thin. It seems to me much thinner than 0.25 mm. Am I wrong?
